docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,225 @@
+
+To test a generator expression, without having to recreate the table, use the
+`random_string(expression)` function:
+
+```sql
+SELECT random_string('#{Name.first_name}')
+```
+
+### Non-character types
+
+Faker supports the following non-character types:
+
+- `BIGINT`
+- `INTEGER` or `INT`
+- `SMALLINT`
+- `TINYINT`
+- `BOOLEAN`
+- `DATE`
+- `DECIMAL`
+- `REAL`
+- `DOUBLE`
+- `INTERVAL DAY TO SECOND`
+- `INTERVAL YEAR TO MONTH`
+- `TIMESTAMP` and `TIMESTAMP(P)`
+- `TIMESTAMP WITH TIME ZONE` and `TIMESTAMP(P) WITH TIME ZONE`
+- `TIME` and `TIME(P)`
+- `TIME WITH TIME ZONE` and `TIME(P) WITH TIME ZONE`
+- `IPADDRESS`
+- `UUID`
+
+Generator expressions cannot be used for non-character-based columns. To limit
+their data range, specify constraints in the `WHERE` clause.
+
+### Unsupported types
+
+Faker does not support the following data types:
+
+- structural types: `ARRAY`, `MAP`, `ROW`
+- `JSON`
+- Geometry
+- HyperLogLog and all digest types
+
+To generate data using these complex types, data from column of primitive types
+can be combined, like in the following example.
+
+```sql
+CREATE TABLE faker.default.prices (
+  currency VARCHAR NOT NULL WITH (generator = '#{Currency.code}'),
+  price DECIMAL(8,2) NOT NULL
+);
+
+SELECT JSON_OBJECT(KEY currency VALUE price) AS complex
+FROM faker.default.prices
+WHERE price > 0
+LIMIT 3;
+```
+
+Executing these queries should return data structured like this:
+```
+      complex
+-------------------
+ {"TTD":924657.82}
+ {"MRO":968292.49}
+ {"LTL":357773.63}
+(3 rows)
+```
+
+### Number of generated rows
+
+By default, the connector generates 1000 rows for every table. To control how
+many rows are generated for a table, use the `LIMIT` clause in the query. A
+default limit can be set using the `default_limit` table, or schema property or
+in the connector configuration file, using the `faker.default-limit` property.
+
+### Null values
+
+For columns without the `NOT NULL` constraint, null values are generated using
+the default probability of 50%. It can be modified using the `null_probability`
+property set for a column, table, or schema. The default value of 0.5 can be
+also modified in the connector configuration file, by using the
+`faker.null-probability` property.
+
+(faker-type-mapping)=
+## Type mapping
+
+The Faker connector generates data itself, so no mapping is required.
+
+(faker-sql-support)=
+## SQL support
+
+The connector provides {ref}`globally available <sql-globally-available>` and
+{ref}`read operation <sql-read-operations>` statements to generate data.
+
+To define the schema for generating data, it supports the following features:
+
+- {doc}`/sql/create-table`
+- {doc}`/sql/create-table-as`
+- {doc}`/sql/drop-table`
+- {doc}`/sql/create-schema`
+- {doc}`/sql/drop-schema`
+
+## Limitations
+
+- Generated data is not deterministic. There is no way to specify a seed for
+  the random generator.
+- All schema objects created in this connector are not persisted, and are
```

Comment:
What does that mean for cluster restarts and also the above determinism ..  this is kinda complex and we need to detail that .. also .. whats the impact of this if I create a large table .. will it crash or prevent me from doing so?

Reply:
It means the schema will not be persisted across cluster restarts, and needs to be recreated every time before using it. I think it's implied by "are not persisted".

Reply:
Thats what I thought .. I assume this is immune to worker restarts and scaling up and down though .. can you confirm and maybe add that info. Also .. maybe we can reword to make it clearer for normal users 

schema objects basically means all the definitions are stored .. and that only in memory

and at the same time .. data is NEVER stored .. its always generated

We have to word it so non-technical users understand and dont get wrong ideas. 

To some degree I think we could even explain that all in the intro rather than as a limitation.. is basically a design and feature that it works like that

Reply:
I moved it to the configuration section.